src/intro-to-c/debugging/general-debugging.md

```diff
@@ -0,0 +1,519 @@
+
+If we want to see all five bananas, we need to use a separate variable for the inner loop. Here, we'll use a variable `j` as its counter.
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+    int i, j;
+    char name[100] = "banana";
+
+    // This should print the name as a grid with 5 rows like:
+    // b a n a n a
+    // b a n a n a
+    // b a n a n a
+    // b a n a n a
+    // b a n a n a
+
+    for (i = 0; i < 5; i++)
+    {
+        for (j = 0; j < 100; j++)
+        {
+            printf("%c ", name[j]);
+        }
+        printf("\n");
+    }
+
+    return 0;
+}
+```
+
+If we run that, we get our expected result:
+
+``` {.terminal}
+b a n a n a                                                                                                                                                                                             
+b a n a n a                                                                                                                                                                                             
+b a n a n a                                                                                                                                                                                             
+b a n a n a                                                                                                                                                                                             
+b a n a n a                                                                                                                                                                                             
+
+```
+
+# Example C: Chaotic Change.
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+    double change = 0.00;
+    int dollars = 0, quarters = 0, dimes = 0, nickels = 0, pennies = 0;
+
+    printf("Enter the amount of change: ");
+    scanf("%lf", &change);
+
+    while (change != 0)
+    {
+        if (change >= 1)
+        {
+            dollars++;
+            change -= 1;
+        }
+        else if (change >= 0.25)
+        {
+            quarters++;
+            change -= 0.25;
+        }
+        else if (change >= 0.10)
+        {
+            dimes++;
+            change -= 0.10;
+        }
+        else if (change >= 0.05)
+        {
+            nickels++;
+            change -= 0.05;
+        }
+        else if (change >= 0.01)
+        {
+            pennies++;
+            change -= 0.01;
+        }
+    }
+
+    printf("%d Dollars\n", dollars);
+    printf("%d Quarters\n", quarters);
+    printf("%d Dimes\n", dimes);
+    printf("%d Nickels\n", nickels);
+    printf("%d Pennies\n", pennies);
+
+    return 0;
+}
+```
+
+``` {.terminal}
+Enter the amount of change: 4.39
+
+```
+
+Why isn't my program printing anything? Maybe something is going wrong inside of the loop? Let me put in a few print statements toward the end of the loop and see what they show.
```

Comment:
How does this sound: "In the `pennies` check, I'll print two decimal places to see how many cents I have left, then at the end of the loop, I'll print 20 decimals places. It's more than I need, but maybe it will show me something I'm missing." I can put it right after that paragraph.

Reply:
Yep that sounds good!